Modules/_datetimemodule.c

```diff
@@ -2284,21 +2284,33 @@ delta_bool(PyDateTime_Delta *self)
 static PyObject *
 delta_repr(PyDateTime_Delta *self)
 {
-    if (GET_TD_MICROSECONDS(self) != 0)
-        return PyUnicode_FromFormat("%s(%d, %d, %d)",
-                                    Py_TYPE(self)->tp_name,
-                                    GET_TD_DAYS(self),
-                                    GET_TD_SECONDS(self),
-                                    GET_TD_MICROSECONDS(self));
-    if (GET_TD_SECONDS(self) != 0)
-        return PyUnicode_FromFormat("%s(%d, %d)",
-                                    Py_TYPE(self)->tp_name,
-                                    GET_TD_DAYS(self),
-                                    GET_TD_SECONDS(self));
-
-    return PyUnicode_FromFormat("%s(%d)",
+    char days[64] = "", seconds[64] = "", microseconds[64] = "";
```

Comment:
I got the 64 bytes idea from [here](https://github.com/musically-ut/cpython/blob/94d5a4e4d33a1d14a2bb1be8fbff5e1e4cd2b7a6/Modules/_datetimemodule.c#L1201), which could also be 6 digits + 1. However, in light of your next comment, we may not end up needing this at all.